code/modules/antagonists/superheroes/buzzon/buzzon_equipment.dm

```diff
@@ -0,0 +1,294 @@
+	icon_state = "bee_full_cryo"
+	hoodtype = /obj/item/clothing/head/hooded/bee_hood/full/cryo
+
+	cold_protection = CHEST|GROIN|ARMS|LEGS
+	min_cold_protection_temperature = FIRE_SUIT_MIN_TEMP_PROTECT
+
+/obj/item/clothing/suit/hooded/bee_costume/buzzon/Initialize()
+	. = ..()
+	allowed = GLOB.security_vest_allowed
+
+/obj/item/clothing/suit/hooded/bee_costume/buzzon/cryo/Initialize()
+	. = ..()
+	allowed = GLOB.security_hardsuit_allowed
+
+/obj/item/clothing/suit/hooded/bee_costume/buzzon/proc/recall_sword()
+	if(!ishuman(loc))
+		return
+	var/mob/living/carbon/human/owner = loc
+	if(locate(/obj/item/melee/beesword/buzzon) in owner.contents)
+		linked_sword = locate() in owner.contents
+		to_chat(owner, "<span class='notice'>You successfully link [linked_sword] to your [src].</span>")
+		return
+
+	if(owner == linked_sword.loc || sword_cooldown > world.time)
+		return
+
+	var/sword_location = get_turf(linked_sword)
+	var/owner_location = get_turf(owner)
+	if(get_dist(owner_location, sword_location) > 10)
+		to_chat(owner,"<span class='warning'>[linked_sword] is too far away!</span>")
+	else
+		sword_cooldown = world.time + 20
+		if(isliving(linked_sword.loc))
+			var/mob/living/current_owner = linked_sword.loc
+			current_owner.dropItemToGround(linked_sword)
+			to_chat(current_owner, "<span class='warning'>[linked_sword]'s small rocket engine suddenly activates and rips it out of your hand!</span>")
+		linked_sword.throw_at(owner, 10, 2)
+
+/obj/item/clothing/head/hooded/bee_hood/full //It's not a helmet because I want tackles and flashbangs to fuck you up
+	name = "cybernetic bee helmet"
+	desc = "A cybernetic helmet attached to a suit. All hail the Queen!"
+	icon_state = "bee_full"
+
+	body_parts_covered = HEAD
+	clothing_flags = THICKMATERIAL
+	flags_inv = HIDEFACE|HIDEHAIR|HIDEFACIALHAIR|HIDEEARS|HIDEEYES|HIDESNOUT
+
+/obj/item/clothing/head/hooded/bee_hood/full/cryo
+	icon_state = "bee_full_cryo"
+
+	cold_protection = HEAD
+	min_cold_protection_temperature = FIRE_SUIT_MIN_TEMP_PROTECT
+
+/obj/item/clothing/head/hooded/bee_hood/full/equipped(mob/user, slot, initial)
+	. = ..()
+	if(!ishuman(user))
+		return
+
+	if(slot != ITEM_SLOT_HEAD)
+		return
+
+	var/mob/living/carbon/human/buzzon = user
+
+	var/datum/atom_hud/hud = GLOB.huds[DATA_HUD_DIAGNOSTIC_ADVANCED]
+	hud.add_hud_to(buzzon)
+	ADD_TRAIT(buzzon, TRAIT_DIAGNOSTIC_HUD, CLOTHING_TRAIT)
+	ADD_TRAIT(buzzon, TRAIT_ROBOTIC_FRIEND, CLOTHING_TRAIT) //BuzzOn is the best roboticist ever so bots don't attack him even if emagged
+
+/obj/item/clothing/head/hooded/bee_hood/full/dropped(mob/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/buzzon = user
+
+	var/datum/atom_hud/hud = GLOB.huds[DATA_HUD_DIAGNOSTIC_ADVANCED]
+	hud.remove_hud_from(buzzon)
+	REMOVE_TRAIT(buzzon, TRAIT_DIAGNOSTIC_HUD, CLOTHING_TRAIT)
+	REMOVE_TRAIT(buzzon, TRAIT_ROBOTIC_FRIEND, CLOTHING_TRAIT)
+
+/obj/item/melee/beesword/buzzon
+	name = "The Stinger"
```

Comment:
you should probably rename the normal bee sword to something else

Reply:
why tho?